src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Objects/Properties.jl

```diff
@@ -101,24 +122,22 @@ function is_smooth(P::AbsProjectiveScheme; algorithm=:default)
   end
 end
 
-function _projective_jacobian_criterion(P::AbsProjectiveScheme)
-  if !(base_ring(P) isa Field)
-    throw(NotImplementedError(
-      :is_smooth,
-      "projective Jacobian criterion only implemented when base ring is a field"
-    ))
-  end
+is_smooth(P::AbsProjectiveScheme{<:Ring, <:MPolyRing}; algorithm::Symbol=:default) = true
+is_smooth(P::AbsProjectiveScheme{<:Ring, <:MPolyLocRing}; algorithm::Symbol=:default) = true
+
+function _projective_jacobian_criterion(P::AbsProjectiveScheme{<:Ring, <:MPolyQuoRing})
   if !(is_equidimensional(P))
     throw(NotImplementedError(
       :is_smooth,
       "projective Jacobian criterion only implemented when scheme is equidimensional"
     ))
   end
-  R = base_ring(homogeneous_coordinate_ring(P))
+  R = ambient_coordinate_ring(P)
   I = defining_ideal(P)
   mat = jacobian_matrix(R, gens(I))
   sing_locus = ideal(R, minors(mat, codim(P)))
-  return dim(sing_locus + I) <= 0
+  irrelevant_ideal = ideal(R, gens(R))
+  return is_one(saturation(sing_locus, irrelevant_ideal))
```

Comment:
I think that you want 
                       sing_locus = ideal(R, minors(mat,codim(P)) + I
Otherwise, you could also get points outside of P, because you never add generators of I in lines 139 and 140, whereas this happened in the deleted line 121.